### PR TITLE
MarketBoardItemListing and MarketBoardItemListingHistory opcodes, clerical stuff

### PR DESCRIPTION
**Pull request by karashiiro**
MarketCity enum is used for more than just MB packets, so I renamed it; added missing semicolon, added actorcontrol comments, and the opcodes

### CODE REVIEW

File: src/common/Network/PacketDef/Zone/ServerZoneDef.h [hunk 1]
```diff
@@ -322,15 +322,7 @@ namespace Sapphire::Network::Packets::Server
...
-      /**
-       * 0x01 Limsa Lominsa
-       * 0x02 Gridania
-       * 0x03 Ul'dah
-       * 0x04 Ishgard
-       * 0x07 Kugane
-       * 0x0A Crystarium
-       */
-      uint8_t retainerCity;
+      Common::City marketCity;
```
Comment (NotAdam) [inline src/common/Network/PacketDef/Zone/ServerZoneDef.h]: this comes from town exd and should just be called `town`

https://github.com/SapphireServer/Sapphire/blob/master/src/common/CommonGen.h#L559-L568

File: src/common/Network/PacketDef/Zone/ClientZoneDef.h [hunk 2]
```diff
@@ -40,9 +40,9 @@ struct FFXIVIpcClientTrigger :
   /* 0004 */ uint32_t param11;
   /* 0008 */ uint32_t param12;
   /* 000C */ uint32_t param2;
-  /* 0010 */ uint32_t param4; // todo: really?
-  /* 0014 */ uint32_t param5;
-  /* 0018 */ uint64_t param3;
+  /* 0010 */ uint32_t param3;
+  /* 0014 */ uint32_t param4;
+  /* 0018 */ uint64_t param5;
```
Comment (NotAdam) [inline src/common/Network/PacketDef/Zone/ClientZoneDef.h]: 99% sure these are used in a specific order elsewhere which aren't in this PR, so renaming them won't do what you intend and will instead break other things

File: src/common/Common.h [hunk 3]
```diff
@@ -993,6 +993,16 @@ namespace Sapphire::Common
     CircularAoEPlaced = 7
   };
 
+  enum City : uint8_t
```
Comment (NotAdam) [inline src/common/Common.h]: use this definition instead: https://github.com/SapphireServer/Sapphire/blob/master/src/common/CommonGen.h#L559-L568

**Comment by karashiiro** [comment]
Fixed the stuff 🐄

**Comment by karashiiro** [comment]
Disregard, these offsets are actually wrong, I'm correcting them now

**Comment by karashiiro** [comment]
Wait, they were correct 🤔 I just have a poor workflow.

**Pull request merged**